geopandas/geodataframe.py

```diff
@@ -880,18 +880,38 @@ def astype(self, dtype, copy=True, errors="raise", **kwargs):
 
     def __xor__(self, other):
         """Implement ^ operator as for builtin set type"""
+        warnings.warn(
+            "'^'' operator will be deprecated. Use 'symmetric_difference' instead.",
```

Comment:
```suggestion
            "'^' operator will be deprecated. Use the 'symmetric_difference' method instead.",
```

(and similar for the others)

Reply:
Thanks.